extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -1116,9 +1207,39 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
 }
 
 /**
- * DSL function for [GeoJsonSource].
+ * DSL function for [GeoJsonSource] performing parsing using background thread.
+ * Immediately returns [GeoJsonSource] with no data set and starts preparing actual data
+ * using a worker thread.
+ *
+ * If using runtime styling:
+ *
+ * loadStyle(style(Style.DARK) {
+ *   +geoJsonSource(id) {
+ *    featureCollection(collection)
+ *   }
+ *   ...
+ * }
+ *
+ * compositing style will be performed correctly under the hood.
+ * [Style.OnStyleLoaded] will be emitted in correct moment of time when all sources are parsed.
+ */
+fun geoJsonSource(
+  id: String,
+  block: GeoJsonSource.Builder.() -> Unit
+): GeoJsonSource = GeoJsonSource.Builder(id) {}.apply(block).build()
+
+/**
+ * DSL function for [GeoJsonSource] performing parsing using background thread.
+ * Immediately returns [GeoJsonSource] with no data set,
+ * fully parsed [GeoJsonSource] is returned in [onGeoJsonParsed] callback.
+ *
+ * Using this method means that it is user's responsibility to proceed with adding layers or
+ * other style objects in [onGeoJsonParsed] callback.
  */
-fun geoJsonSource(id: String, block: GeoJsonSource.Builder.() -> Unit): GeoJsonSource =
-  GeoJsonSource.Builder(id).apply(block).build()
+fun geoJsonSource(
+  id: String,
+  block: GeoJsonSource.Builder.() -> Unit,
+  onGeoJsonParsed: (GeoJsonSource) -> Unit
```

Comment:
shall we make the onGeoJsonParsed null by default?
```suggestion
  onGeoJsonParsed: ((GeoJsonSource) -> Unit)? = null
```

Reply:
I was trying it, sadly breaks all our existing code if using it that way.

Reply:
ok, now we have two different methods for geoJsonSource, so I guess we can leave out the default value, as it will have generate same Java signature like
```
fun geoJsonSource(  id: String,  block: GeoJsonSource.Builder.() ->; Unit): GeoJsonSource
```